### PR TITLE
Garbage collector e2e can create too many pods for cluster

### PR DESCRIPTION
**Pull request by smarterclayton**
Smaller clusters don't have enough room to run 100 pods in some cases.
Do our best.

### CODE REVIEW

File: test/e2e/apimachinery/garbage_collector.go [hunk 1]
```diff
@@ -44,6 +44,27 @@ import (
 	imageutils "k8s.io/kubernetes/test/utils/image"
 )
 
+// estimateMaximumPods estimates how many pods the cluster can handle
+// with some wiggle room, to prevent pods being unable to schedule due
+// to max pod constraints.
+func estimateMaximumPods(c clientset.Interface) int32 {
+	availablePods := int32(0)
+	for _, node := range framework.GetReadySchedulableNodesOrDie(c).Items {
```
Comment (liggitt) [inline test/e2e/apimachinery/garbage_collector.go]: this can return 0. what happens then?

File: test/e2e/apimachinery/garbage_collector.go [hunk 2]
```diff
@@ -44,6 +44,27 @@ import (
...
+		if q, ok := node.Status.Allocatable["pods"]; ok {
+			if num, ok := q.AsInt64(); ok {
+				availablePods += int32(num) * 8 / 10
+				continue
+			}
+		}
+		// best guess
+		availablePods += 8
```
Comment (liggitt) [inline test/e2e/apimachinery/garbage_collector.go]: just because?

Comment (smarterclayton) [reply test/e2e/apimachinery/garbage_collector.go]: 10 is the default per core.

File: test/e2e/apimachinery/garbage_collector.go (hunk 1 above)
Comment (smarterclayton) [reply test/e2e/apimachinery/garbage_collector.go]: I'll put a minimum bar

**Comment by smarterclayton** [comment]
Updated with a little bit more context

**Pull request merged**